src/app/store/machine/utils/hooks.ts

```diff
@@ -20,6 +21,24 @@ import {
   isNodeStorageConfigurable,
 } from "app/store/utils";
 import vlanSelectors from "app/store/vlan/selectors";
+import { isId } from "app/utils";
+
+/**
+ * Get a machine via the API.
+ * @param id - A machine's system id.
+ */
+export const useGetMachine = (id?: Machine[MachineMeta.PK] | null): void => {
```

Comment:
This is a very simple hook right now, but it will need to do a lot such as generating a request id and cleaning up the machine in the store once it is no longer required.